### ISSUE TITLE
Client using local scaling can fail to resize (especially with a maximized window)

### ISSUE THREAD

**Issue by marc-legendre** [novnc/noVNC#1903]
Hi there!

I stumbled upon an issue where a client using local scaling can fail to resize when the window resizes. This happens in very specific conditions, for example when *starting* with a maximized window, then resizing it, then maximizing it again.

This happens since 0ff0844a142c086128cce8771a3dad21a499f02b (first released in 1.3.0)

### Reproduction

This is easy to reproduce with the VNC app:

* open the VNC app
* maximize the window
* connect
* resize window
  * the viewer resizes correctly
* maximize the window again
  * *the viewer does not resize*

Here is a screencast demonstrating the issue, using `vnc.html` from master (047531e8), with Chromium (but this is not browser-specific).

[Screencast 2024-09-30 11:40:32.webm](https://github.com/user-attachments/assets/99f3b9f5-512e-4146-a566-87bde981733b)

### Misc. details

I have not completely understood the issue, but I can share some observations / intuitions

* the client resize happens only if `!_clientHasExpectedSize()`
* but `_expectedClientWidth` and `_expectedClientHeight` never change when I resize the window.
* thus if the new window size happens to be the same as the initial size, the client won't resize
* this is why the issue is easily triggered with a maximized window
* this is also why it rarely happens otherwise, and even then it is not even noticeable

**Comment by marc-legendre** [comment]
I don't understand everything in `RFB`, but I am wondering if maybe  `_handleResize()` should call `_saveExpectedClientSize()` like `_resize()` does ?

```patch
diff --git i/core/rfb.js w/core/rfb.js
index 0007bed..2758a1e 100644
--- i/core/rfb.js
+++ w/core/rfb.js
@@ -730,6 +730,7 @@ export default class RFB extends EventTargetMixin {
         window.requestAnimationFrame(() => {
             this._updateClip();
             this._updateScale();
+            this._saveExpectedClientSize();
         });
 
         if (this._resizeSession) {
```

Indeed, both call `_updateClip()` and `_updateScale()`, but only `_resize` calls `_saveExpectedClientSize()`, and it *feels* like `_handleResize` should.

**Comment by CendioOssman** [comment]
Thanks for your bug report! I can reproduce the issue here as well.

I think we didn't fully consider all scenarios in #1617, and we need to reevaluate the design here.

@samhed, can you have a look?

**Comment by samhed** [comment]
Fixed in e6e03a2
